Deprecated: `Fitcontribution` camel case methods

cadenmyers13 opened the pull request:
6 methods of `FitContribution` are deprecated here. See commit messages for which. I pytest passed locally and fitbulkNi works (and throws Deprecation message).

cadenmyers13 commented:
@sbillinge jk... weird error from diffpy.utils. Maybe we need a release of utils first. Still feels odd that its saying no module named `diffpy.utils` instead of not being able to find `_deprecator`.

```
/usr/share/miniconda/envs/test/lib/python3.13/site-packages/diffpy/srfit/fitbase/fitcontribution.py:32: in <module>
    from diffpy.utils._deprecator import deprecated
E   ModuleNotFoundError: No module named 'diffpy.utils'
```

sbillinge commented on news/fitcontribution-dep.rst:
```diff
@@ -0,0 +1,23 @@
+
+**Deprecated:**
+
+* Deprecate camel case method formatting in ``FitContribution``.
```
please can you list all hte public facing methods that were depracated as a matter of policy.  This is user facing so it is more useful to them than which module from a dev point of view that you worked on.

cadenmyers13 commented:
@sbillinge okay i changed some things a bit. First off (as recommended) I changed one test for `setProfile` to have the old version. This tests passes and pytest also throws the deprecation warning message which is great. see screeshot:
<img width="1068" height="301" alt="Screenshot 2025-12-12 at 10 56 50 AM" src="https://github.com/user-attachments/assets/3debae19-af61-47c2-9db6-495348459138" />

Secondly, I added the deprecation message to diffpy.utils in this PR https://github.com/diffpy/diffpy.utils/pull/367. The process for deprecation is now, 

1) In top level `__init__.py`, define a `_REMOVAL_VERSION` variable which is the version where the deprecated function will be removed
2) Go to the module where deprecation will take place and import the removal version, and deprecation tools. Also, define a base variable that references what will be removed. ex,
```
from diffpy.srfit import _REMOVAL_VERSION
from diffpy.utils._deprecator import deprecated, deprecation_message

_BASE_FITCONTRIBUTION = __name__ + ".FitContribution"
```
3) Use `@deprecated` to deprecate a method/function/etc,
```
    @deprecated(
        deprecation_message(
            _BASE_FITCONTRIBUTION,
            "setProfile",
            "set_profile",
            _REMOVAL_VERSION,
        )
    )
    def setProfile(self, profile, xname=None, yname=None, dyname=None):
```

Once again, I tested this with `fitBulkNi.py` and it worked. see screenshot:
<img width="1139" height="217" alt="Screenshot 2025-12-12 at 11 05 34 AM" src="https://github.com/user-attachments/assets/b8f487d4-2f3d-42de-9a36-648b887d9c11" />

cadenmyers13 commented:
@sbillinge okay, i duplicated it. Both versions passed locally

cadenmyers13 commented:
closing to standardize deprecation process
